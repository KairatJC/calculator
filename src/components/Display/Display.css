.calc {
    margin-top: 150px;
}

.calc-display {
    word-break: break-all;
}

.calc-display span {
    margin-right: 5px;
}
