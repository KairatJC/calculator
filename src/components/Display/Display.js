import React from "react";
import "./Display.css";
import CalcContext from "../../CalcContext";

function Display() {
  return (
    <CalcContext.Consumer>
      {value => {
        let result = getResult(value);
        return (
          <div className="calc wrap">
            {result}
          </div>
        );
      }}
    </CalcContext.Consumer>
  );
}

function getResult(value) {
  if (value.operator !== null || value.oldValue !== null) {
    return (
      <div className="calc-display">
        <span>
          {value.oldValue}
        </span>
        <span>
          {value.operator}
        </span>
        <span>
          {value.currentValue}
        </span>
      </div>
    );
  } else {
    return (
      <div className="calc-display">
        <span>
          {value.currentValue}
        </span>
      </div>
    );
  }
}

export default Display;
