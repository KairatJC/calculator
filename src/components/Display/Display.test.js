import React from "react";
import renderer from "react-test-renderer";
import Display from "./Display";

test("Calculator display should render correctly", () => {
  const component = renderer.create(<Display />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
