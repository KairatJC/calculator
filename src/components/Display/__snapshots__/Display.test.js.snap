// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator display should render correctly 1`] = `
<div
  className="calc wrap"
>
  <div
    className="calc-display"
  >
    <span />
  </div>
</div>
`;
